contracts/DCAStrategies/DCAStrategies/DCAStrategiesPermissionsHandler.sol

```diff
@@ -2,51 +2,105 @@
 pragma solidity >=0.8.7 <0.9.0;
 
 import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+import '@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol';
 import '../../interfaces/IDCAStrategies.sol';
+import '../../libraries/PermissionMath.sol';
+import '../../utils/Governable.sol';
 
-abstract contract DCAStrategiesPermissionsHandler is IDCAStrategiesPermissionsHandler, ERC721 {
-  /// @inheritdoc IDCAStrategiesPermissionsHandler
-  // solhint-disable-next-line func-name-mixedcase
-  function PERMIT_TYPEHASH() external pure override returns (bytes32) {}
+abstract contract DCAStrategiesPermissionsHandler is IDCAStrategiesPermissionsHandler, ERC721, EIP712, Governable {
+  struct TokenPermission {
+    // The actual permissions
+    uint8 permissions;
+    // The block number when it was last updated
+    uint248 lastUpdated;
+  }
 
-  /// @inheritdoc IDCAStrategiesPermissionsHandler
-  // solhint-disable-next-line func-name-mixedcase
-  function PERMISSION_PERMIT_TYPEHASH() external pure override returns (bytes32) {}
+  using PermissionMath for IDCAStrategies.Permission[];
+  using PermissionMath for uint8;
 
   /// @inheritdoc IDCAStrategiesPermissionsHandler
-  // solhint-disable-next-line func-name-mixedcase
-  function PERMISSION_SET_TYPEHASH() external pure override returns (bytes32) {}
-
+  bytes32 public constant PERMIT_TYPEHASH = keccak256('Permit(address spender,uint256 tokenId,uint256 nonce,uint256 deadline)');
   /// @inheritdoc IDCAStrategiesPermissionsHandler
-  // solhint-disable-next-line func-name-mixedcase
-  function DOMAIN_SEPARATOR() external view override returns (bytes32) {}
-
-  /// @inheritdoc IERC721BasicEnumerable
-  function totalSupply() external view override returns (uint256) {}
-
+  bytes32 public constant PERMISSION_PERMIT_TYPEHASH =
+    keccak256(
+      'PermissionPermit(PermissionSet[] permissions,uint256 tokenId,uint256 nonce,uint256 deadline)PermissionSet(address operator,uint8[] permissions)'
+    );
+  /// @inheritdoc IDCAStrategiesPermissionsHandler
+  bytes32 public constant PERMISSION_SET_TYPEHASH = keccak256('PermissionSet(address operator,uint8[] permissions)');
   /// @inheritdoc IDCAStrategiesPermissionsHandler
   // TODO: update this after building the new descriptor
-  function nftDescriptor() external override returns (IDCAHubPositionDescriptor) {}
+  IDCAHubPositionDescriptor public nftDescriptor;
+  /// @inheritdoc IDCAStrategiesPermissionsHandler
+  mapping(address => uint256) public nonces;
+  mapping(uint256 => uint256) public lastOwnershipChange;
+  mapping(uint256 => mapping(address => TokenPermission)) public tokenPermissions;
```

Comment:
I think it would be better if instead of using a nest mapping, we did something like [this](https://github.com/Mean-Finance/dca-v2-periphery/blob/main/contracts/DCAFeeManager/DCAFeeManager.sol#L21)

We are basically hashing a two part key into one, so we don't need to read storage twice